rpc/src/rpc.rs

```diff
@@ -1002,26 +1002,36 @@ impl JsonRpcRequestProcessor {
         })
     }
 
-    fn check_blockstore_root<T>(
+    // Check if the given `slot` is within the blockstore bounds. This function assumes that
+    // `result` is from a blockstore fetch, and that the fetch:
+    // 1) Checked if `slot` is above the lowest cleanup slot (and errored if not)
+    // 2) Checked if `slot` is a root
+    fn check_blockstore_bounds<T>(
         &self,
         result: &std::result::Result<T, BlockstoreError>,
         slot: Slot,
     ) -> Result<()> {
-        if let Err(err) = result {
-            debug!(
-                "check_blockstore_root, slot: {:?}, max root: {:?}, err: {:?}",
-                slot,
-                self.blockstore.max_root(),
-                err
-            );
-            if slot >= self.blockstore.max_root() {
-                return Err(RpcCustomError::BlockNotAvailable { slot }.into());
-            }
-            if self.blockstore.is_skipped(slot) {
-                return Err(RpcCustomError::SlotSkipped { slot }.into());
+        match result {
+            // The slot was found, all good
+            Ok(Some(_)) => Ok(()),
```

Comment:
Hmm, of the two Blockstore methods, only `get_rooted_block_time()` returns a `Result<Option<T>>` (the other is `Result<VersionedBlock>`). Should we edit `get_rooted_block_time()` to map None to some Blockstore error?

Reply:
Hmm, looking at instances of `fn get_rooted_`, 3/4 return a `Result<Option<T>>` with `get_rooted_block()` being the only one that doesn't return an option. So, I think we stick with `Result<Option<T>>` and I can do something in the caller to make the the passed in type compatible

Reply:
After sleeping on this, my thoughts have changed a little 😅 . The method `get_block_time()` (and similar) should definitely return a `Result<Option<T>>`
- The `Result` captures potential blockstore errors
- The `Option` captures the key/value pair potentially being absent

On the other hand, I think it could be reasonable for `get_rooted_block_time()` to return just a `Result<T>`
- The `Result` still captures potential blockstore errors
- But, the value being absent is arguably an error

> Should we edit get_rooted_block_time() to map None to some Blockstore error?

So yeah, you were seemingly thinking the same thing earlier and I'm now leaning more that way

Reply:
> So yeah, you were seemingly thinking the same thing earlier and I'm now leaning more that way

Yes, exactly! Sorry I didn't articulate it as well and thoroughly as you did :)

Reply:
Addressing in https://github.com/solana-labs/solana/pull/33955 - we can continue the discussion there so closing this out